src/movies.js

```diff
@@ -1,35 +1,80 @@
 // The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
+// console.log('movies: ', movies);
 
+const movies = require('./data');
 
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(array) {
+  let allDirectors = array.map(function (movies) {
+    return movies.director;
+  });
+  return allDirectors;
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+
+function howManyMovies(movies) {
+  return movies.filter((movie) => {
+    return (
+      movie.genre.includes('Drama') && movie.director === 'Steven Spielberg'
+    );
+  }).length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+
+const scoresAverage = (arr) => {
+  if (arr.length === 0) return 0;
+  let allScoresAvg = arr.reduce(function (acc, value) {
+    if (!value.score) return acc;
+
+    return acc + value.score;
+  }, 0);
+  return Number((allScoresAvg / arr.length).toFixed(2));
+};
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
 
+function dramaMoviesScore(moviesArr) {
+  const dramaMovies = moviesArr.filter((movie) =>
+    movie.genre.includes('Drama')
+  );
+  const avgDramaMoviesScore = scoresAverage(dramaMovies);
+  return avgDramaMoviesScore;
+}
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear(array) {
+  let arrayNew = [];
+  array.sort((a, b) => {
+    if (a.year < b.year) return -1;
+    if (a.year > b.year) return 1;
+    if (a.year === b.year) return 0;
+  });
```

Comment:
Here try and use the a.something - b.something for the sort, furthermore, you should use localeCompare instead of returning -1/0/1